src/coll/mod.rs

```diff
@@ -185,11 +187,17 @@ where
         options: impl Into<Option<DropCollectionOptions>>,
         session: impl Into<Option<&mut ClientSession>>,
     ) -> Result<()> {
+        let session = session.into();
+
         let mut options = options.into();
         resolve_options!(self, options, [write_concern]);
 
-        let drop = DropCollection::new(self.namespace(), options);
-        self.client().execute_operation(drop, session.into()).await
+        let drop = DropCollection::new(
+            self.namespace(),
+            options,
+            transaction_read_concern(&session),
```

Comment:
In order to avoid having to do this for each new write operation that we introduce, we can include this in `execute_operation`. To do so, we'd need a helper on `Command` that we could check in the event that the transaction state is starting, and if the command contains no read concern but the transaction does, set it there.